Kernel/Memory/MemoryManager.cpp

```diff
@@ -383,6 +387,100 @@ UNMAP_AFTER_INIT void MemoryManager::parse_memory_map()
     });
 }
 
+UNMAP_AFTER_INIT void MemoryManager::parse_memory_map_efi(MemoryManager::GlobalData& global_data)
+{
+    VERIFY(g_boot_info.boot_method == BootMethod::EFI);
+
+    dmesgln("MM: EFI memory map:");
+    for (size_t i = 0; i < g_boot_info.boot_method_specific.efi.memory_map.descriptor_array_size; i += g_boot_info.boot_method_specific.efi.memory_map.descriptor_size) {
+        auto const* descriptor = bit_cast<EFI::MemoryDescriptor const*>(bit_cast<FlatPtr>(g_boot_info.boot_method_specific.efi.memory_map.descriptor_array) + i);
+
+        auto length = descriptor->number_of_pages * EFI::EFI_PAGE_SIZE;
+        auto start_paddr = PhysicalAddress { descriptor->physical_start };
+        auto end_paddr = PhysicalAddress { descriptor->physical_start + length };
+
+        static constexpr Array memory_type_names = {
+            "Reserved"sv,
+            "LoaderCode"sv,
+            "LoaderData"sv,
+            "BootServicesCode"sv,
+            "BootServicesData"sv,
+            "RuntimeServicesCode"sv,
+            "RuntimeServicesData"sv,
+            "Conventional"sv,
+            "Unusable"sv,
+            "ACPIReclaim"sv,
+            "ACPI_NVS"sv,
+            "MemoryMappedIO"sv,
+            "MemoryMappedIOPortSpace"sv,
+            "PALCode"sv,
+            "Persistent"sv,
+            "Unaccepted"sv,
+        };
+
+        static constexpr size_t max_memory_type_name_length = []() {
+            size_t max_length = 0;
+            for (auto name : memory_type_names)
+                max_length = max(name.length(), max_length);
+            return max_length;
+        }();
+
+        if (to_underlying(descriptor->type) < memory_type_names.size())
+            dmesgln("  {}-{}: {:<{}} attributes={:p}", start_paddr, end_paddr, memory_type_names[to_underlying(descriptor->type)], max_memory_type_name_length, to_underlying(descriptor->attribute));
```

Comment:
edk2 ovmf has around 100 memory map entries, which causes this to be quite spammy. Most of these entries are for boot services, so they also get printed when MM logs all avaiable physical regions.